quivr/tables.py

```diff
@@ -960,6 +961,29 @@ def validate(self) -> None:
             except errors.ValidationError as e:
                 raise errors.ValidationError(f"Column {name} failed validation: {str(e)}", e.failures) from e
 
+    def invalid_mask(self) -> pa.Array:
+        """Return a boolean mask indicating which rows are invalid."""
+        num_rows = self.table.num_rows
+        mask = np.zeros(num_rows, dtype=bool)
+        for name, validator in self._column_validators.items():
+            indices, _ = validator.failures(self.table.column(name))
+            mask[indices.to_numpy()] = True
```

Comment:
Numpy just makes things so easy sometimes. I was curious to see how you were going to do this with pyarrow compute functions.

Reply:
I implemented it both ways and this appeared the more efficient (both compute and memory) of the two.